### ISSUE TITLE
Anonymous top module confuses Agda

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#1388]
As expected, true is not in scope when trying to bind it to 'test':

``` agda
module _ where

open import Common.Equality

module _ where
  open import Common.Prelude

test = true  -- fails as expected
```

However, if I delete the top-level anonymous module header, it is:

``` agda
open import Common.Equality

module _ where
  open import Common.Prelude

test = true  -- works, but should fail
```

I would expect a parse error, as first stuff is indented, and then not.  
However, it seems to be legal Agda:

``` agda
module IndentationInTopModule where

  indented = Set

not-indented = Set

  -- again-indented = Set  -- Parse error
```

This parses:  you can indent in the top module, then unindent.  But if you 
indent again, you get a parse error!?

Note that this is not allowed in inner modules!

``` agda
module InnerModule where

    ind = Set

  not = Set  -- fails

    again-indented = Set
```

My opinion:  Also in the top level module, if you start indented, you need to 
continue indented.  There should not be a special treatment of the top module 
other than you can omit the indentation altogether.

Original issue reported on code.google.com by `andreas....@gmail.com` on 8 Dec 2014 at 2:34

**Comment by andreasabel** [comment]
Issue rediscovered by a user in
https://lists.chalmers.se/pipermail/agda/2016/008463.html

**Comment by andreasabel** [comment]
It seems that the weird behavior is introduced by the following code in `Parser.y`:

``` haskell
-- | Insert a top-level module if there is none.
figureOutTopLevelModule :: [Declaration] -> [Declaration]
figureOutTopLevelModule ds =
  case span isAllowedBeforeModule ds of
    (ds0, Module r m tel ds1 : ds2) -> ds0 ++ [Module r m tel $ ds1 ++ ds2]
    (ds0, ds1)                      -> ds0 ++ [Module (getRange ds1) (QName noName_) [] ds1]
```

case 1: If there are no non-import statements before the first module, we add the declarations `ds2` following the first module to the first module declarations `ds1`.  This makes code like

``` agda
module File where
  ds1
ds2
```

legal when it really should be a parse error.  In general, either `ds1` or `ds2` should be empty for this to be legal.

**Comment by andreasabel** [comment]
Issue was introduced by https://github.com/agda/agda/commit/f30432239881536f92f50bc048307453ca528279 which fixed #953.
